internal/sbi/producer/pdu_session.go

```diff
@@ -60,6 +60,10 @@ func HandlePDUSessionSMContextCreate(request models.PostSmContextsRequest) *http
 	smContext.SMLock.Lock()
 	defer smContext.SMLock.Unlock()
 
+	upi := smf_context.GetUserPlaneInformation()
+	upi.Mu.RLock()
```

Comment:
Read lock is being used for session creation because it seems that multiple session creations are allowed in respect to user_plane_information consistency. Same comment for `HandlePDUSessionSMContextUpdate`.